pkg/orchestration/wiring/kubeingress/plugin.go

```diff
@@ -232,15 +265,34 @@ func buildIngressHostName(resourceName voyager.ResourceName, sc wiringplugin.Sta
 		clusterHostPath)
 }
 
-func extractKubeComputeDependency(dependencies []wiringplugin.WiredDependency) (*apps_v1.Deployment, bool /* retriable */, error) {
+func extractSingleDependencyOfType(dependencies *[]wiringplugin.WiredDependency, resourceType voyager.ResourceType) (*wiringplugin.WiredDependency, bool /* retriable */, error) {
+	var matchedDependency *wiringplugin.WiredDependency
+	for x := range *dependencies {
```

Comment:
you can just do `for _, dependency := range *dependencies` here and then you don't need to reference by index below.

Reply:
I copy-pasted this from elsewhere in the codebase (including the index access). Will change, thanks.

Reply:
Done.

Reply:
See my comment =)

Reply:
Yeah, lovely. Fixed.